File: lib/data/models/notification_item.dart
// A notification entry shown in the Notifications feature and used by the
// home screen's "Alertas recientes" summary block.

enum NotificationKind {
  newBoard,
  indicatorThresholdExceeded,
  newReport,
  whistleblowerAlert,
}

class NotificationItem {
  final String id;
  final NotificationKind kind;
  final String title;
  final String body;
  final DateTime receivedAt;
  // Opaque target identifier (e.g., a report id, a board id, a case id).
  // The UI maps it to a navigation action.
  final String? targetRef;
  final bool read;

  const NotificationItem({
    required this.id,
    required this.kind,
    required this.title,
    required this.body,
    required this.receivedAt,
    this.targetRef,
    this.read = false,
  });
}

File: Data/Models/NotificationItem.cs
namespace OrquestadorFrontend.Data.Models;

public enum NotificationKind
{
    NewBoard,
    IndicatorThresholdExceeded,
    NewReport,
    WhistleblowerAlert
}

// A notification entry shown in the Notifications feature and used by the home
// screen's "Alertas recientes" summary block.
public sealed record NotificationItem(
    string Id,
    NotificationKind Kind,
    string Title,
    string Body,
    DateTime ReceivedAt,
    string? TargetRef = null,
    bool Read = false);
